### PR TITLE
Implementation of state root provider contract.

### LINKED ISSUE #175: Implement StateRootProvider
1. Take existing `Anchor` logic to make a `StateRootProvider is MosaicVersion`
    - Create storage variables
    - Create internal function `_anchorStateRoot`

### PR DESCRIPTION
**Pull request by deepesh-kn**
This PR contains the following changes:
1. Created a new solidity file `StateRootProvider`.
2. Added most of the code from `Anchor.sol`.
3. Removed `ConsensusModule`/`CoconsensusModule` references.
4. Made the functions `setup` and `anchorStateRoot` as internal.

Fixes: #175

### CODE REVIEW

File: contracts/anchor/StateRootProvider.sol [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+contract StateRootProvider is CircularBufferUint {
+
+    /* Usings */
+
+    using SafeMath for uint256;
+
+
+    /* Events */
+
+    event StateRootAvailable(uint256 _blockHeight, bytes32 _stateRoot);
```
Comment (benjaminbollen) [inline contracts/anchor/StateRootProvider.sol]: `_blockNumber`

File: contracts/anchor/StateRootProvider.sol [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+
+    /* Storage */
+
+    /** Maps block heights to their respective state root. */
```
Comment (benjaminbollen) [inline contracts/anchor/StateRootProvider.sol]: "block numbers"

Comment (benjaminbollen) [reply contracts/anchor/StateRootProvider.sol]: and later occurances

**Review by benjaminbollen — CHANGES_REQUESTED** [review]
please update naming `block height` to `block number`

File: contracts/anchor/StateRootProvider.sol (hunk 2 above)
Comment (deepesh-kn) [reply contracts/anchor/StateRootProvider.sol]: Done

File: contracts/anchor/StateRootProvider.sol (hunk 1 above)
Comment (deepesh-kn) [reply contracts/anchor/StateRootProvider.sol]: Done

**Review by benjaminbollen — APPROVED** [review]
LGTM

**Pull request merged**